src/gui/qgsabstractdatasourcewidget.h

```diff
@@ -86,6 +86,13 @@ class GUI_EXPORT QgsAbstractDataSourceWidget : public QDialog
     //! Emitted when a vector layer has been selected for addition
     void addVectorLayer( const QString &uri, const QString &layerName );
 
+    /** Emitted when one or more OGR supported layers are selected for addition
+     * \param layerQStringList list of layers protocol URIs
+     * \param enc encoding
+     * \param dataSourceType string (can be "file" or "database")
+     */
+    void addVectorLayers( const QStringList &layerQStringList, const QString &enc, const QString &dataSourceType );
```

Comment:
Can we change  `layerQStringList` to just `layerList` ? layerQStringList is a bit clumsy to me. Also enc should be encoding - see https://github.com/qgis/qgis3.0_api/issues/53